ImperatorToCk2/src/main/java/ImperatorToCK2/Importer.java

```diff
@@ -1326,5 +1329,364 @@ public static String[] importSaveInfo (String directory) throws IOException //im
+                        if (embColor2.equals("noColor2")) {
+                            embColor2 = "none";
+                        }
+                        if (embTexture.equals("noTexture")) {
+                            embTexture = "none";
+                        }
+                        if (instanceYes == 0) { //if emblem has no instances specified, generate instance
+                            if (output[4].equals("0")) {//build emblem
+                                output[4] = embTexture+"~_~"+embColor1+"~_~"+embColor2+"~_~"+embScale+"~_~"+embPos+"~_~"+embRot;
+                            } else {
+                                output[4] = output[4]+"~~"+embTexture+"~_~"+embColor1+"~_~"+embColor2+"~_~"+embScale+"~_~"+embPos+"~_~"+embRot;
+                            }
+                        } else {
+                            tmpOutput = tmpOutput.replace("noColor1",embColor1);
+                            tmpOutput = tmpOutput.replace("noColor2",embColor2);
+                            tmpOutput = tmpOutput.replace("noTexture",embTexture);
+                            if (output[4].equals("0")) { //build emblem
+                                output[4] = tmpOutput;
+                            } else {
+                                output[4] = output[4]+"~~"+tmpOutput;
+                            }
+                        }
+                    }
+
+                    if (qaaa.equals( "}" ) ) { //ends here
+
+                        String[] tmpOutput = new String[output.length];
+                        int aq2 = 0;
+                        while (aq2 < output.length) {
+                            tmpOutput[aq2] = output[aq2];
+                            aq2 = aq2 + 1;
+                        }
+
+                        impFlagList.add(tmpOutput);
+
+                        output[0] = "unnamedFlag"; //default for no tag/name
+                        output[1] = "pattern_solid.tga"; //default for no pattern
+                        output[2] = "none"; //default for color1, format = hsvOrRgb~r,g,b
+                        output[3] = "none"; //default for color2, format = hsvOrRgb~r,g,b
+                        output[4] = "0"; //default for no emblems, format is texture~_~color1~_~color2~_~scale~_~position~_~rotation~~(nextEmblem)
+
+                    }
+
+                }
+
+                //}
+            }
+
+        }catch (java.util.NoSuchElementException exception){
+            endOrNot = false;
+
+        }   
+
+        return impFlagList;
+
+    }
+
+    public static ArrayList<String[]> importColors (String name) throws IOException //named_colors for flags
+    {
+        String VM = "\\";
+        VM = VM.substring(0);
+        name = name+VM+"game"+VM+"common"+VM+"named_colors"+VM+"default_colors.txt";
+        
+        FileInputStream fileIn= new FileInputStream(name);
+        Scanner scnr= new Scanner(fileIn);
+
+        ArrayList<String[]> impColorList= new ArrayList<String[]>();
```

Comment:
If you return a map,
```
Map<String, Color> impColorList= new HashMap<String, Color>();
```

Reply:
I haven't heard of the Map data type before, I might try it out in the future, but I don't know enough about it currently to use it for this PR